app/src/main/java/org/sagebionetworks/research/mpower/signup/ResponseErrorCode.java

```diff
@@ -0,0 +1,82 @@
+
+import android.support.annotation.IntDef;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+//
+//  ResponseErrorCode.java
+//  mPower2
+//
+//  Copyright © 2018 Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+/**
+ * This interface defines the various types of actions that are possible to have.
+ */
+@Retention(RetentionPolicy.SOURCE)
+@IntDef({ResponseErrorCode.UNKNOWN, ResponseErrorCode.NOT_SET_UP_ERROR,
+        ResponseErrorCode.INTERNET_NOT_CONNECTED, ResponseErrorCode.SERVER_NOT_REACHABLE,
+        ResponseErrorCode.SERVER_UNDER_MAINTENAINCE, ResponseErrorCode.SERVER_NOT_AUTHENTICATED,
+        ResponseErrorCode.PRECONDITION_NOT_MET, ResponseErrorCode.NO_CREDENTIALS_AVAILABLE,
+        ResponseErrorCode.UNSUPPORTED_APP_VERSION, ResponseErrorCode.SERVER_NOT_AUTHORIZED,
+        ResponseErrorCode.SERVER_ACCOUNT_DISABLED, ResponseErrorCode.S3_UPLOAD_ERROR_RESPONSE,
+        ResponseErrorCode.NOT_A_FILE_URL, ResponseErrorCode.NOT_EXPECTED_CLASS,
+        ResponseErrorCode.TEMP_FILE_ERROR, ResponseErrorCode.FILE_READ_ERROR,
+        ResponseErrorCode.OBJECT_NOT_FOUND, ResponseErrorCode.NOT_A_VALID_SURVEY_REF,
+        ResponseErrorCode.NOT_A_VALID_JSON_OBJECT, ResponseErrorCode.NOT_REGISTERED_FOR_PUSH_NOTIFICATIONS})
+public @interface ResponseErrorCode {
```

Comment:
I know we have our own way in Java/Android of doing response codes through exception types, but I thought I'd post this concept about doing it through a standard generic type Response that contains an error code that maps the same on iOS.